subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -964,6 +967,10 @@ func generateAffinityGroupBindInfo(
 					// in its "nodes" and "gpuIndices" as the node and GPU address
 					if mbi.PodPlacements[podIndex].PhysicalNode == "" {
 						mbi.PodPlacements[podIndex].PhysicalNode = nodes[0]
+						// check if all pods are scheduled to suggested nodes for a new group
+						if group == nil && !suggestedNodeSet.Contains(nodes[0]) {
+							allInSuggested = false
```

Comment:
>allInSuggested [](start = 7, length = 14)

It only means allInSuggested for new group. For old group, it can be true, but may not all in suggested.
So, rename it to "newGroupAllInSuggested ?

Reply:
Can we move "if group == nil" to caller, so allInSuggested can be kept here for simple

---
In reply to: [387525219](https://github.com/microsoft/pai/pull/4251#discussion_r387525219) [](ancestors = 387525219)